File: antibytes-publishing/src/test/java/tech/antibytes/gradle/publishing/ClosureHelper.java
/*
 * Copyright (c) 2022 Matthias Geisler (bitPogo) / All rights reserved.
 *
 * Use of this source code is governed by Apache License, Version 2.0
 */

package tech.antibytes.gradle.publishing;

import com.palantir.gradle.gitversion.VersionDetails;

import groovy.lang.Closure;

class ClosureHelper {
    static public Closure<VersionDetails> createClosure(VersionDetails value) {
        return new Closure<VersionDetails>(value) {
            public VersionDetails doCall(Object args) {
                return value;
            }
        };
    }
}

File: antibytes-gradle-test-utils/src/main/kotlin/tech/antibytes/gradle/test/ClosureHelper.kt
/*
 * Copyright (c) 2024 Matthias Geisler (bitPogo) / All rights reserved.
 *
 * Use of this source code is governed by Apache v2.0
 */

@file:Suppress("ktlint:standard:filename")

package tech.antibytes.gradle.test

import groovy.lang.Closure

fun <T> createClosure(value: T): Closure<T> {
    return object : Closure<T>(value) {
        fun doCall(args: Any?): T = value
    }
}
